File: Java/프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int answer = 0;
        while (storey > 0) {
            int remaind = storey % 10;
            int nextDigit = (storey / 10) % 10;
            
            if (remaind > 5) {
                answer += (10 - remaind);
                storey = (storey / 10) + 1;
            } else if (remaind < 5) {
                answer += remaind;
                storey /= 10;
            } else {
                answer += 5;
                if (nextDigit >= 5) {
                    storey = (storey / 10) + 1;
                } else {
                    storey /= 10;
                }
            }
        }
        return answer;
    }
}

File: Swift/프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.swift
import Foundation

func solution(_ storey:Int) -> Int {
    var current = storey
    var result = 0
    while current > 0 {
        let remaind = current % 10
        let nextDigit = (current / 10) % 10
        
        if remaind > 5 {
            result += (10 - remaind)
            current = (current / 10) + 1
        } else if remaind < 5 {
            result += remaind
            current /= 10
        } else {
            result += 5
            if nextDigit >= 5 {
                current = (current / 10) + 1
            } else {
                current /= 10
            }
        }
    }
    return result
}